### PR TITLE
SYCL. Add more test for sycl-objectives

### PR DESCRIPTION
**Pull request by razdoburdin**
This PR adopts more tests for sycl-objectives.

### CODE REVIEW

File: src/common/ranking_utils.h [hunk 1]
```diff
@@ -320,7 +320,9 @@ class NDCGCache : public RankingCache {
   }
 
   linalg::VectorView<double const> InvIDCG(Context const* ctx) const {
-    return inv_idcg_.View(ctx->Device());
+  // This function doesn't have sycl-specific implementation yet.
+  // For that reason we transfer data to host in case of sycl is used for propper execution.
+    return inv_idcg_.View(ctx->Device().IsSycl() ? DeviceOrd::CPU() : ctx->Device());
```
Comment (trivialfis) [inline src/common/ranking_utils.h:325]: Feel free to make suggestion on how we should structure the project in the future.

**Review by trivialfis — COMMENTED** [review]
Please help fix some of the compilation failures when you are available.

**Comment by razdoburdin** [comment]
> Please help fix some of the compilation failures when you are available.
I have fixed linking issues, but some cuda tests still faille

**Comment by razdoburdin** [comment]
Hi @trivialfis, could you please relaunch cuda-ci? It looks like the failures are not related to PR content.

**Comment by trivialfis** [comment]
Just saw it, restarted.

**Comment by razdoburdin** [comment]
> Just saw it, restarted.
it was some problem with the CI configuration 
image
I have merged the master branch, hope it would help

**Pull request merged**